### PR TITLE
[test][PAC][AArch64] Add ELF tests for subtarget-neutral codegen

### PR DESCRIPTION
**Pull request by kovdan01**
Many parts of PAuth-related codegen are not MachO- or ELF-specific. Add RUN lines against ELF targets to ensure that codegen works for ELF as well as for MachO.

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/ptrauth-bti-call.ll [hunk 1]
```diff
@@ -1,101 +1,128 @@
...
+; RUN:   -o - %s | FileCheck %s --check-prefixes=DARWIN,CHECK
+; RUN: llc -mtriple aarch64-linux-gnu   -mattr=+bti -mattr=+pauth              -asm-verbose=false \
+; RUN:   -o - %s | FileCheck %s --check-prefixes=ELF,CHECK
+; RUN: llc -mtriple aarch64-linux-gnu   -mattr=+bti -mattr=+pauth -global-isel -asm-verbose=false \
+; RUN:   -o - %s | FileCheck %s --check-prefixes=ELF,CHECK
+; RUN: llc -mtriple aarch64-linux-gnu   -mattr=+bti -mattr=+pauth -fast-isel   -asm-verbose=false \
+; RUN:   -o - %s | FileCheck %s --check-prefixes=ELF,CHECK
 
-; ptrauth tail-calls can only use x16/x17 with BTI.
+;; ptrauth tail-calls can only use x16/x17 with BTI.
```
Comment (asl) [inline llvm/test/CodeGen/AArch64/ptrauth-bti-call.ll:14]: do we need this change? :)

File: llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll [hunk 2]
```diff
@@ -1,44 +1,58 @@
...
   %tmp = call i64 @llvm.ptrauth.blend(i64 %arg, i64 %arg1)
   ret i64 %tmp
 }
 
 define i64 @test_blend_constant(i64 %arg) {
-; CHECK-LABEL: test_blend_constant:
-; CHECK:       ; %bb.0:
-; CHECK-NEXT:    movk x0, #12345, lsl #48
-; CHECK-NEXT:    ret
+; DARWIN-LABEL: _test_blend_constant:
```
Comment (asl) [inline llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll:23]: Do we really need to test the platform differences? Can we use `CHECK-LABEL: test_blend_constant` and `CHECK: %bb.0` on all platforms? Same in other files.

**Review by asl — CHANGES_REQUESTED** [review]
I think the set of changes could be reduced by not checking irrelevant platform differences

File: llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll (hunk 2 above)
Comment (kovdan01) [reply llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll:23]: > Do we really need to test the platform differences? Can we use `CHECK-LABEL: test_blend_constant` and `CHECK: %bb.0` on all platforms? Same in other files.

Yes, we can. I'll follow this approach, thanks

File: llvm/test/CodeGen/AArch64/ptrauth-bti-call.ll (hunk 1 above)
Comment (kovdan01) [reply llvm/test/CodeGen/AArch64/ptrauth-bti-call.ll:14]: It looks like that new tests use a convention: for "special" lines like RUN, CHECK, etc, use ';' prefix, and for "actual comments" - use ';;' prefix. I've not found this in official guidelines, but it's clear from existing tests and I also feel that there were PRs where this was discussed.

If the verb in the question is **need**, then no, we don't need that, it's just a cosmetic change.

File: llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll (hunk 2 above)
Comment (kovdan01) [reply llvm/test/CodeGen/AArch64/ptrauth-intrinsic-blend.ll]: I've got rid of checking platform differences. See 41faccabd93ae6ef528ab4c4da04e1a9062d582e

**Review by asl — APPROVED** [review]

**Pull request merged**